### PR TITLE
View Crossposts

### PR DESCRIPTION
**Pull request by Sjmarf**
Lets you view (not create) crossposts of a post. Relies on [this MlemMiddleware PR](https://github.com/mlemgroup/MlemMiddleware/pull/62).

### CODE REVIEW

**Comment by Sjmarf** [comment]
Updated to use the design discussed on Slack 👍

File: Mlem.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -2691,8 +2695,8 @@
 			isa = XCRemoteSwiftPackageReference;
 			repositoryURL = "https://github.com/mlemgroup/MlemMiddleware";
 			requirement = {
-				kind = upToNextMinorVersion;
-				minimumVersion = 0.38.0;
+				branch = sjmarf/post3;
+				kind = branch;
 			};
```
Comment (EricBAndrews) [inline Mlem.xcodeproj/project.pbxproj]: Reminder

File: Mlem/App/Views/Shared/ExpandedPost/ExpandedPostView.swift [hunk 2]
```diff
@@ -39,19 +39,24 @@ struct ExpandedPostView: View {
...
-                    content(post: post, tracker: tracker, isLoading: proxy.isLoading)
+                if let post = proxy.entity {
+                    content(post: post, isLoading: proxy.isLoading)
                         .externalApiWarning(entity: post, isLoading: proxy.isLoading)
-                        .task {
-                            if post.api == appState.firstApi, tracker.loadingState == .idle {
-                                post.markRead()
+                        .task(id: tracker == nil) {
+                            if let tracker, post.api == appState.firstApi, tracker.loadingState == .idle {
+                                // post.markRead()
```
Comment (EricBAndrews) [inline Mlem/App/Views/Shared/ExpandedPost/ExpandedPostView.swift]:
```suggestion
                                post.markRead()
```

**Review by EricBAndrews — APPROVED** [review]
LGTM!

Posts don't look like they're getting marked read when opened--there's a line below that I think got commented out accidentally?

File: Mlem/App/Views/Shared/ExpandedPost/ExpandedPostView.swift (hunk 2 above)
Comment (Sjmarf) [reply Mlem/App/Views/Shared/ExpandedPost/ExpandedPostView.swift]: Nice catch!

**Pull request merged**